crates/consensus/src/transaction/typed.rs

```diff
@@ -307,6 +336,61 @@ impl Transaction for OpTypedTransaction {
     }
 }
 
+impl SignableTransaction<Signature> for OpTypedTransaction {
+    fn set_chain_id(&mut self, chain_id: ChainId) {
+        self.delegate_signable_mut(|tx| tx.set_chain_id(chain_id)).unwrap_or(())
+    }
+
+    fn set_chain_id_checked(&mut self, chain_id: ChainId) -> bool {
+        self.delegate_signable_mut(|tx| tx.set_chain_id_checked(chain_id)).unwrap_or(true)
+    }
+
+    fn encode_for_signing(&self, out: &mut dyn alloy_rlp::BufMut) {
+        self.delegate_signable(|tx| tx.encode_for_signing(out)).unwrap_or(())
+    }
+
+    fn payload_len_for_signature(&self) -> usize {
+        self.delegate_signable(|tx| tx.payload_len_for_signature()).unwrap_or(0)
+    }
+
+    fn encoded_for_signing(&self) -> Vec<u8> {
+        self.delegate_signable(|tx| tx.encoded_for_signing()).unwrap_or_default()
+    }
+
+    fn signature_hash(&self) -> B256 {
+        self.delegate_signable(|tx| tx.signature_hash()).unwrap_or(B256::ZERO)
+    }
+
+    fn into_signed(self, signature: Signature) -> Signed<Self, Signature>
+    where
+        Self: Sized,
+    {
+        match self {
+            Self::Deposit(tx) => Signed::new_unchecked(
```

Comment:
yeah,
given that we also don't this trait implemented for the ethereum typed enum and only for the actual raw transactions, this feels a bit odd, because a `signed<TxDeposit>` isn't useful.

can't think of a usecase where we'd need `TypedTx -> Signed`, 
I assume the proper way to do this would be to convert it back to the `TxRequest` and then "build"/sign it